src/test/java/io/reactivex/schedulers/TestSchedulerTest.java

```diff
@@ -216,4 +217,40 @@ public void run() {
             inner.dispose();
         }
     }
+
+    @Test
+    public final void testConcurrentInsertionAndTimeChange() {
+        final TestScheduler scheduler = new TestScheduler();
+        new Thread() {
+            @Override public void run() {
+                while (true) {
+                    Thread.yield(); // Scheduling service lock is not fair
+                    final Scheduler.Worker inner = scheduler.createWorker();
+                    inner.schedule(new Runnable() {
+                        @Override
+                        public void run()
+                        {
+                            System.out.println(scheduler.now(TimeUnit.MILLISECONDS));
+                        }
+
+                    },1, TimeUnit.SECONDS);
+                }
+            }
+        }.start();
+        try
+        {
+            for (int i = 0; i < 10; i++) {
+                Thread.sleep(5);
+                scheduler.advanceTimeBy(1, TimeUnit.SECONDS);
+            }
+        } catch (InterruptedException e)
+        {
+            e.printStackTrace();
```

Comment:
Just add `throws InterruptedException` to method signature, it should fail the test, I guess